### PR TITLE
Add news functions

### PR DESCRIPTION
**Pull request by theo-bnts**
News methods :
    listSets()
    getReplayDownloadLink()
    getWeaponsDetails()
    listWeaponsSpawnChance()
    getGameModeExtendedData()
    listFeaturedCreativeIslands()
    searchIsland()
    listFish()
    getPlayerFishStats()

### CODE REVIEW

File: lib/endpoints.js [hunk 1]
```diff
@@ -47,5 +48,15 @@ module.exports = {
     listUsersById: (ids = []) => `${endpoint}/lookupUsername?id=${ids.join()}`,
     getBundles: lang => `${endpoint}/bundles?lang=${lang}`,
     listLoot: lang => `${endpoint}/loot/list?lang=${lang}`,
-    getLootDetails: (id, lang) => `${endpoint}/loot/get?id=${id}&lang=${lang}`
+    getLootDetails: (id, lang) => `${endpoint}/loot/get?id=${id}&lang=${lang}`,
+
```
Comment (benhawley7) [inline lib/endpoints.js]:
```suggestion
```
nitpick: Whitespace

File: lib/fortnite-api.js [hunk 2]
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
...
+
+    /**
+     * listSets()
+     * List all the sets used by cosmetics.
+     *
+     * @param {Object} [options]
+     * @param {String} [options.lang] en, ar, de, es, es-419, fr, it, ja, ko, pl, pt-BR, ru, tr
+     * @returns {Promise}
+     */
+
```
Comment (benhawley7) [inline lib/fortnite-api.js]: nitpick: whitespace
```suggestion
```

File: lib/fortnite-api.js [hunk 3]
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
...
+        return this.request(uri);
+    }
+
+    /**
+     * listFeaturedCreativeIslands()
+     * List the current featured islands in creative mode
+     *
+     * @returns {Promise}
+     */
+
```
Comment (benhawley7) [inline lib/fortnite-api.js]:
```suggestion
```

nitpick: whitespace

File: lib/fortnite-api.js [hunk 4]
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
...
+    /**
+     * listFish()
+     * Get the list of fish (name, description, image) as well as their minimum and maximum length
+     *
+     * @param {Object} [options]
+     * @param {String} [options.lang] unknown supported languages
+     * @returns {Promise}
+     */
+
```
Comment (benhawley7) [inline lib/fortnite-api.js]:
```suggestion
```

nitpick: whitespace

File: package.json [hunk 5]
```diff
@@ -1,34 +1,65 @@
...
+    "saveSpec": null,
+    "fetchSpec": "^1.5.2"
+  },
+  "_requiredBy": [
+    "/"
+  ],
+  "_resolved": "https://registry.npmjs.org/fortnite-api-io/-/fortnite-api-io-1.5.2.tgz",
+  "_shasum": "e2be1d06036dc26b5723d4c529d0a8430397866c",
+  "_spec": "fortnite-api-io@^1.5.2",
+  "_where": "D:\\Documents\\Bots Discord\\Bot CMaker",
```
Comment (benhawley7) [inline package.json]: This stuff shouldn't be here?

File: lib/fortnite-api.js [hunk 6]
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
...
+     * getWeaponsDetails()
+     * Get all stats for a specific loot/weapon item
+     * 
+     * @premium
+     * @param {String} id of weapon
+     * @param {Object} [options]
+     * @param {String} [options.lang] en, ar, de, es, es-419, fr, it, ja, ko, pl, pt-BR, ru, tr
+     * @returns {Promise}
+     */
+    getWeaponsDetails(id, options = {}) {
```
Comment (benhawley7) [inline lib/fortnite-api.js:463]: I think this should be `getWeaponDetails()` since its one weapon!

File: lib/fortnite-api.js [hunk 7]
```diff
@@ -417,6 +421,127 @@ class FortniteAPI {
...
+
+    /**
+     * listWeaponsSpawnChance()
+     * List the spawn chances for each type for a given game mode.
+     *
+     * @premium
+     * @param {String} mode
+     * @returns {Promise}
+     */
+    listWeaponsSpawnChance(mode) {
```
Comment (benhawley7) [inline lib/fortnite-api.js]: Likewise, I would say this should be `listWeaponSpawnChances()`!

**Review by benhawley7 — CHANGES_REQUESTED** [review]
Thanks for the PR @theo-bnts !

Lots of good work for the endpoints!

Just some changes to be made for method names and random whitespace in the file!

There are also some strange properties in the package.json which need removing!

File: lib/fortnite-api.js (hunk 7 above)
Comment (theo-bnts) [reply lib/fortnite-api.js]: I thinking that correct.

**Review by theo-bnts — COMMENTED** [review]
> Thanks for the PR @theo-bnts !
> 
> Lots of good work for the endpoints!
> 
> Just some changes to be made for method names and random whitespace in the file!
> 
> There are also some strange properties in the package.json which need removing!

No problem !

That good, I have patch all :)

File: lib/fortnite-api.js (hunk 7 above)
Comment (benhawley7) [reply lib/fortnite-api.js]: I would say `listWeaponSpawnChances` is the most appropriate name for this method, given the description in the API docs.

**Review by benhawley7 — APPROVED** [review]
Looks good! Thanks for updating the endpoint and also adding the supported languages for each method in the JSDoc.

Before we merge, I think its best we rename `listWeaponsSpawnChance` to `listWeaponSpawnChances` as this reads better to me and is closer to what is written in the API Docs.

A tip for the future is to write more descriptive commit messages. At the moment they are usually something like `Update fortnite-api.js` which doesn't explain what the change to the code is. Some examples could be:

`feature: Added listFIsh method to the client class`
`fix: Added missing language option to the getTournaments method`

(I like to prefix the the message with `fix:` or `feature:` so I know if its a bug fix or not)

File: lib/fortnite-api.js (hunk 7 above)
Comment (theo-bnts) [reply lib/fortnite-api.js]: Okay, all is good :)

**Comment by theo-bnts** [comment]
> Looks good! Thanks for updating the endpoint and also adding the supported languages for each method in the JSDoc.
> 
> Before we merge, I think its best we rename `listWeaponsSpawnChance` to `listWeaponSpawnChances` as this reads better to me and is closer to what is written in the API Docs.
> 
> A tip for the future is to write more descriptive commit messages. At the moment they are usually something like `Update fortnite-api.js` which doesn't explain what the change to the code is. Some examples could be:
> 
> `feature: Added listFIsh method to the client class`
> `fix: Added missing language option to the getTournaments method`
> 
> (I like to prefix the the message with `fix:` or `feature:` so I know if its a bug fix or not)

No problem, I have patched `listWeaponsSpawnChance` to `listWeaponSpawnChances` and I will add more informations on commit messages in the future 👍

**Pull request merged**